homeassistant/components/tplink/__init__.py

```diff
@@ -103,17 +184,56 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
 
     hass.data[DOMAIN][entry.entry_id] = TPLinkDataUpdateCoordinator(hass, device)
+    # sdb9696: Should await? from the docs: You don’t want to await this coroutine if it is called as part of the setup of a component, because it can cause a deadlock.
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
 
 
+config_entry_syncer_id: Optional[str] = None
+
+
+async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+    global config_entry_syncer_id
+    if (
+        entry.options.get(TPLINK_CLOUD_CREDENTIALS_SYNC)
+        and config_entry_syncer_id is None
+    ):
+        config_entry_syncer_id = entry.entry_id
+        try:
+            for config_entry in hass.config_entries.async_entries(DOMAIN):
+                if entry != config_entry and config_entry.options.get(
+                    TPLINK_CLOUD_CREDENTIALS_SYNC
+                ):
+                    if config_entry.options.get(
+                        TPLINK_CLOUD_USERNAME
+                    ) != entry.options.get(
+                        TPLINK_CLOUD_USERNAME
+                    ) or config_entry.options.get(
+                        TPLINK_CLOUD_PASSWORD
+                    ) != entry.options.get(
+                        TPLINK_CLOUD_PASSWORD
+                    ):
```

Comment:
It would be easier to just update them all without checking if a change is needed, given the syncing is enabled.